### DISCUSSION TITLE
création d'un composite qui change en fonction d'un état

### DISCUSSION THREAD

**Discussion by Kabech** [witsa/synapps#727] [Q&A]
bonjour;
je veut crée un composite qui change en fonction de l'État exemple :
j ai un capteur de présence si le capteur détecte une Personne je met une photo avec un œil si non un œil barrée.
est ce que c est par un code ou évènement ?
comment je peut en remédier a ca svp ?
![1](https://github.com/witsa/synapps/assets/169897242/8f967d91-919a-4178-a4dd-e2785705ae59)

merci de vos retour.

**Comment by JeremieDENAEGEL** [comment]
Bonjour,
Je t'invite à utiliser l'acteur "image interrupteur" à la place de l'acteur "image".
Cela sera plus simple par rapport à ton besoin.

**Comment by Kabech** [comment]
Merci pour le retour,
Est ce que vous me conseillez de faire comme le composi vanne ? 
Pour récupérer l'état d'une lampe et enfin la contrôler depuis le plan !!

**Comment by WITNicolas** [comment]
Petit possibilité sans l'acteur image et en utillisant les glyphs présentent dans Studio.
C'est un acteur html-icon
Selon le même principe, il est possible de modifier dynamiquement les couleurs (ou en fait n'importe quelle property)

![image](https://github.com/witsa/synapps/assets/33632664/da1d3c58-d9a4-4614-be9c-96d94fca317f)


```text
SYNAPPS-STUDIO-ACTOR|{"type":"display/html","key":"iconChange","properties":{"content":"<i class=\"icon-{{icon}}\"></i>","top":"610px","left":"490px"},"additionalDefs":{"icon":{"type":"icon","label":"Icône","description":"Web Hosting Hub Glyphs"},"propOn":{"type":"icon","label":"Icône Etat Vrai","description":"Web Hosting Hub Glyphs","helperTxt":"Choisir Icone (vide pour rien)"},"propOff":{"type":"icon","label":"Icône Etat Faux","description":"Web Hosting Hub Glyphs","helperTxt":"Choisir Icone (vide pour rien)"},"propVariable":{"type":"boolean","label":"Variable","helperTxt":"Lier une variable"}},"additionals":{"icon":"eye-open","propOn":"eye-open","propOff":"eye-close","propVariable":true},"events":{"onPropertyChanged":["this.additionals.icon = this.additionals.propVariable ? this.additionals.propOn : this.additionals.propOff"]}}

**Reply by belkasse** [reply]
Bonjour Nicolas, je suis débutant sur synapps 
A quoi servent ces scripts ou ligne de commande que l’on voit souvent dans les discussions ?

**Reply by WITNicolas** [reply]
Bonjour,
Il s'agit de script que l'on peut copier et coller dans la section apropriée dans STUDIO.
Si commence par SYNAPPS-STUDIO-ACTOR| c'est donc un acteur que l'on peut coller dans une scene.
Si SYNAPPS-STUDIO-COMPOSITE| alors il faut le coller dans la section composite etc...
SYNAPPS-STUDIO-SCENE ..... et bien c'est un scene.

Dans tous les cas, il s'agit de créations, de modeles d'objets que la communauté souhaite partager.

Tout simplement,

**Comment by Kabech** [comment]
parfait; merci beaucoup.
bonne journée

**Comment by belkasse** [comment]
Merci 🙏


***@***.***
***@***.******@***.******@***.******@***.******@***.***<https://www.instagram.com/coriance/?hl=fr>

Samir BELKACEM
Responsable d’exploitation
+33 (0)5 61 16 87 21
+33 (0)6 73 64 13 57
Agence Sud-Ouest
1 Chemin de Perpignan
31100 Toulouse
05 61 16 87 20





________________________________
De : WITNicolas ***@***.***>
Envoyé : jeudi 13 juin 2024 19:00
À : witsa/synapps ***@***.***>
Cc : BELKACEM Samir ***@***.***>; Comment ***@***.***>
Objet : Re: [witsa/synapps] création d'un composite qui change en fonction d'un état (Discussion #727)

ATTENTION: L'expéditeur de ce mail est externe à Coriance. Soyez vigilant lorsque vous devez cliquer sur des liens ou ouvrir des pièces jointes. En cas de doute, contactez le Support Informatique.


Bonjour,
Il s'agit de script que l'on peut copier et coller dans la section apropriée dans STUDIO.
Si commence par SYNAPPS-STUDIO-ACTOR| c'est donc un acteur que l'on peut coller dans une scene.
Si SYNAPPS-STUDIO-COMPOSITE| alors il faut le coller dans la section composite etc...
SYNAPPS-STUDIO-SCENE ..... et bien c'est un scene.

Dans tous les cas, il s'agit de créations, de modeles d'objets que la communauté souhaite partager.

Tout simplement,

—
Reply to this email directly, view it on GitHub<https://github.com/witsa/synapps/discussions/727#discussioncomment-9765859>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BI7J2J2PLVW6D6SQ6XM3HQDZHHFZ7AVCNFSM6AAAAABH2H3JXWVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TONRVHA2TS>.
You are receiving this because you commented.Message ID: ***@***.***>

*** Ce message est confidentiel et destiné uniquement à la personne ou à l’entité à laquelle il est adressé. Toute utilisation, retransmission, reproduction ou autre de ce message ou de son contenu par des personnes autres que le destinataire est interdite. Si vous avez reçu ce message par erreur, nous vous remercions de nous en informer et de le supprimer de votre système informatique. This message is confidential and intended only for the person or entity to which it is addressed. Any review, copy, disclosure, retransmission, dissemination or other use of this information by persons or entities other than the intended recipient is prohibited. If you have received it by mistake, please let us know by reply and then delete it from your system. ***
